compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java

```diff
@@ -164,6 +171,56 @@ public static boolean isValidPng(Path filePath) throws IOException {
         return isMatchingImageFormat(filePath, PNG_HEX_HEADER, 8);
     }
 
+    /**
+     * Get last modified timestamp of a ballerina project.
+     *
+     * @param projectRoot project root path
+     * @return last modified time of the ballerina project
+     */
+    public static long lastModifiedTimeOfBalProject(Path projectRoot) {
+        File[] files = projectRoot.toFile().listFiles();
+        long latestDate = 0;
+        if (files != null) {
+            for (File file : files) {
+                long fileModifiedDate = latestDate;
+                String filename = file.toPath().getFileName().toString();
+                if (file.isDirectory()) {
+                    if (isChild(projectRoot.resolve(TEST_DIR_NAME), file.toPath())
+                            && isChild(projectRoot.resolve(RESOURCE_DIR_NAME), file.toPath())) {
+                        fileModifiedDate = lastModifiedTimeOfBalProject(file.toPath());
+                    } else if (file.toPath().equals(projectRoot.resolve(MODULES_ROOT))) {
+                        // modules directory, not considering files inside
+                        fileModifiedDate = file.lastModified();
+                    }
+                } else {
+                    if (file.toPath().equals(projectRoot.resolve(BALLERINA_TOML))) {
```

Comment:
`CompilerPlugin.toml` also needs to be checked

Reply:
Added